SPECS/openssl/openssl.spec

```diff
@@ -162,38 +164,32 @@ export HASHBANGPERL=/usr/bin/perl
     enable-camellia \
     no-capieng \
     enable-cast \
-    no-chacha \
     enable-cms \
-    no-comp \
     enable-ct \
     enable-deprecated \
     enable-des \
     enable-dh \
     enable-dsa \
     no-dtls1 \
-    no-ec2m \
     enable-ec_nistp_64_gcc_128 \
     enable-ecdh \
     enable-ecdsa \
     no-gost \
-    no-idea \
     no-mdc2 \
     no-md2 \
     enable-md4 \
-    no-poly1305 \
     enable-rc2 \
     enable-rc4 \
     enable-rc5 \
-    no-rfc3779 \
     enable-rmd160 \
     no-sctp \
-    no-seed \
     no-siphash \
     no-sm2 \
-    no-sm3 \
-    no-sm4 \
     no-ssl \
     no-ssl3 \
+    enable-ssl3-method \
```

Comment:
are these needed just to make the python-cryptography tests pass? If so, it actually is probably better to keep openssl as it is, and modify/skip the python-cryptography tests to handle the algorithms we already have.

Reply:
these were the changes i needed to get python-cryptography to build and get the module to load for tests.

we could try to find a way for the rust code not to need them, i didn't have much luck initially, but it is probably possible.

Reply:
i can disable the py-crypto bits that fail without the openssl changes by patching py-crypto.

failures start showing up in tpm2-pytss tests with our existing openssl.  i can probably disable those tests too, though hard to know how important they are (though tpm2-pytss is only needed for swtpm, which is a test tool for us).

Reply:
I think that would be best

Reply:
done ... lots of tests disabled via patches, but everything builds and remaining tests pass :)